app/src/main/java/com/example/admin/personallibrarycatalogue/AddBookActivityFragment.java

```diff
@@ -0,0 +1,191 @@
+            }
+        });
+
+        // Actions for ok - button (Checking fields, insert into database, start new activity)
+        Button okButton = (Button) rootView.findViewById(R.id.apply_button);
+
+        okButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                EditText authorEditText = (EditText) rootView.findViewById(R.id.author_edit_text);
+                String author = authorEditText.getText().toString();
+
+                EditText titleEditText = (EditText) rootView.findViewById(R.id.title_edit_text);
+                String title = titleEditText.getText().toString();
+
+                EditText descriptionEditText = (EditText) rootView.findViewById(R.id.description_edit_text);
+                String description = descriptionEditText.getText().toString();
+
+                ImageView coverView = (ImageView) rootView.findViewById(R.id.cover_image_view);
+                byte[] cover = Util.getBytesFromDrawable(coverView.getDrawable());
+
+                if (author.matches("")) {
+                    Toast.makeText(getActivity(), PLEASE_ENTER_THE_AUTHOR_FIELD, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                if (title.matches("")) {
+                    Toast.makeText(getActivity(), PLEASE_ENTER_THE_TITLE_FIELD, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                Book book = new Book(id_,title, author, description, cover);
+
+                // in case user edit information just update data
+                if (id_ != null) {
+                    helper.updateBook(id_, book);
+                } else {
+                    helper.addBook(book);
+                }
+
+                Intent intent = new Intent(getActivity(), BooksListActivity.class);
+                startActivity(intent);
+
+            }
+        });
+
+        // Actions for cancel button (clear fields and leave activity)
+        Button cancelButton = (Button) rootView.findViewById(R.id.cancel);
+        cancelButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                ViewGroup group = (ViewGroup) rootView.findViewById(R.id.add_book_layout);
+                clearForm(group);
+
+                Intent intent = new Intent(getActivity(), MainActivity.class);
+                startActivity(intent);
+            }
+        });
+
+        return rootView;
+    }
+
+    /**
+     * Clears all edtiText fileds
+     */
+    private void clearForm(ViewGroup group) {
+        for (int i = 0, count = group.getChildCount(); i < count; i++) {
+            View view = group.getChildAt(i);
+            if (view instanceof EditText) {
+                ((EditText) view).setText("");
+            }
+        }
+    }
+
+    private void fillAllFields(View view, LibraryDatabaseHelper helper, int id) {
```

Comment:
No need to pass ```LibraryDatabaseHelper``` here as argument. LibraryDatabaseHelper should be Singleton with synchronization and be accessed everywhere in the app.